build-and-test.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -e 	# Exit immediately upon failure
+set -o pipefail  # Carry failures over pipes
+
+repo_root="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+docker_repo="microsoft/dotnet-nightly"
+
+pushd "${repo_root}" > /dev/null
+
+find . -path './.git' -prune -o -path './test' -prune -o -path './update-dependencies' -prune -o -name 'Dockerfile' -printf '%p\t%d\n' | egrep -v 'nanoserver|windowsservercore' | sort -n -k2 | while read line
```

Comment:
The `test` folder may have Dockerfiles in the `artifacts` subfolder created by `run-test.sh`, so we should at least exclude that one. I think we can safely assume that the `.git` folder won't have Dockerfiles, but I think that the `update-dependencies` folder could in the future, so I decided to just exclude them all.  I can remove the exclusion of the `.git` folder, but it seems like it might be good to still explicitly exclude the `update-dependencies` folder.  What are your thoughts on that?